File: brn.hs
import Control.Monad
import System.Directory

main = do
  bd <- getCurrentDirectory
  xs <- getDirectoryContents bd
  let dirs = filter (`notElem` [".", ".."]) xs
  forM_ dirs $ \dir -> do
    fs <- getDirectoryContents dir
    let files = filter (`notElem` [".", ".."]) fs
    forM_ files $ \file -> do
      let fop = bd ++ "/" ++ dir ++ "/" ++ file
      let fnp = bd ++ "/" ++ dir ++ " - " ++ file
      renameFile fop fnp

File: brn.lua
local lfs = require("lfs")

bd = lfs.currentdir()
for dir in lfs.dir(bd) do
   if dir ~= "." and dir ~= ".." then
      for file in lfs.dir(dir) do
         if file ~= "." and file ~= ".." then
            fop = string.format("%s/%s/%s", bd, dir, file)
            fnp = string.format("%s/%s - %s", bd, dir, file)
            os.rename(fop, fnp)
         end
      end
   end
end
